arangod/Aql/Functions.cpp

```diff
@@ -7475,3 +7476,61 @@ AqlValue Functions::NotImplemented(ExpressionContext* expressionContext,
   registerError(expressionContext, "UNKNOWN", TRI_ERROR_NOT_IMPLEMENTED);
   return AqlValue(AqlValueHintNull());
 }
+
+AqlValue Functions::GetValidation(ExpressionContext* expressionContext,
+                              transaction::Methods* trx,
+                              VPackFunctionParameters const& parameters) {
+
+  static char const* AFN = "GET_VALIDATION"; // GET_VALIDATON("collectionName")
+
+  if(parameters.size() != 1) {
+    THROW_ARANGO_EXCEPTION_PARAMS(TRI_ERROR_QUERY_FUNCTION_ARGUMENT_NUMBER_MISMATCH, AFN);
+  }
+
+  std::string const collectionName = ::extractCollectionName(trx, parameters, 0);
+
+  if (collectionName.empty()){
+    THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_BAD_PARAMETER, "could not extract collection name from parameters");
+  }
+
+  auto logicalCollection = expressionContext->vocbase().lookupCollection(collectionName);
+  if(!logicalCollection) {
+    THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND, "could not find collection");
+  }
+
+  VPackBuilder builder;
+  logicalCollection->validatorsToVelocyPack(builder); // returns object but should return Array
+  return AqlValue(builder);
+}
+
+AqlValue Functions::Validate(ExpressionContext* expressionContext,
+                              transaction::Methods* trx,
+                              VPackFunctionParameters const& parameters) {
+
+  static char const* AFN = "VALIDATE"; // VALIDATE(doc, [ validator ] )
+
+  if(parameters.size() != 2) {
+    THROW_ARANGO_EXCEPTION_PARAMS(TRI_ERROR_QUERY_FUNCTION_ARGUMENT_NUMBER_MISMATCH, AFN);
+  }
+
+  AqlValue const& docValue = extractFunctionParameterValue(parameters, 0);
+  AqlValue const& schemaValue = extractFunctionParameterValue(parameters, 1);
+
+  if(!schemaValue.isObject()){
+    THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_BAD_PARAMETER, "no object given validation objects");
```

Comment:
this error message is misleading. "no object given validation objects" should rather be "expecting schema object for schema validation" or such.